### DISCUSSION TITLE
Can a dynamic topic model be saved too?

### DISCUSSION THREAD

**Discussion by DominikTrier** [MaartenGr/BERTopic#931] [Q&A] [answered]
Hi folks,

I want to save a GPU-trained model and load it on a CPU-workstation for visualization. The normal models works just fine but I want to export a Dynamic Topic Model too. If I call visualize_topics_over_time() on the CPU workstation, I need to recompute the topics_over_time() model but this takes an awful amount of time on a CPU. Is there a way to save this DTM model like the normal model?
Any help is appreciated!

**Comment by MaartenGr** [comment, marked as answer]
When you run `.topics_over_time` the model is not internally saving information with respect to the dynamic topic model but returns **all** information you need as a dataframe. In other words, when you run the following:

```python
topics_over_time = topic_model.topics_over_time(tweets, timestamps, nr_bins=20)
```

The information with respect to the dynamic nature of topics is contained in the DataFrame `topics_over_time`. This means that you would only need to save `topics_over_time` as a csv locally and pass it later on to `.visualize_topics_over_time`.

**Reply by DominikTrier** [reply]
Thank you very much for the quick and clear answer! Very appreciated!
